### ISSUE TITLE
[Placement Group] Ray Tune + distributed training support

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#9919]
tl;dr - colocation for gpu workers in distributed training is important. 

## Goals
Allow users to easily colocate workers for distributed tuning/training.

## Explicit non-goals
No need to support odd worker configurations or custom placement strategies.

## Proposed API
```python
DistributedTrainableCreator(
   use_gpu=False,
   num_hosts=1,  # or num_nodes
   num_slots=1,  # or workers_per_node
   num_cpus_per_slot=1,
   timeout_s=30):
    """Configuration for distributed training scheduling.

    One basic assumption of
    this implementation is that all sub-workers
    of a trial will be placed evenly across different machines.

    It is recommended that if `num_hosts` per trial > 1, you set
    num_slots == the size (or number of GPUs) of a single host.
    If num_hosts == 1, then you can set num_slots to be <=
    the size (number of GPUs) of a single host.


    Args:
        use_gpu (bool); Whether to allocate a GPU per worker.
        num_cpus_per_slot (int): Number of CPUs to request
            from Ray per worker.
        num_hosts (int): Number of hosts that each trial is expected
            to use.
        num_slots (int): Number of slots (workers) to start on each host.
```
## Implementation notes:

Ideally we use the new placement group API to implement colocation.
```python
pgs = []
all_workers = []
rank = 0
for node in num_nodes:
    pgs.append(PlacementGroup.create(
        {"CPU": num_cpus_per_worker * num_workers,
        "GPU": num_workers * use_gpu},
    ))
    for i in num_workers:
        workers.append(
            TrainableWorker.options(
               placement_group=pgs[-1]).remote(rank=rank))
        rank += 1
```

Some caveats:
* No autoscaling
* Upon failure, placement group needs to be explicitly deleted

**Comment by richardliaw** [comment]
https://github.com/ray-project/ray/issues/9799 duplicate?

**Comment by ericl** [comment]
Ah, this is the experimental integration which can be done earlier. Though we can dedup

**Comment by ericl** [comment]
This is blocked on https://github.com/ray-project/ray/issues/9800

**Comment by rkooo567** [comment]
https://github.com/ray-project/ray/pull/11061

cc @richardliaw Will there be any other future PRs to close this issue other than colocation?

**Comment by richardliaw** [comment]
Yeah, this won’t be closed yet, we will need to refactor some other parts of Tune that handle resource management.

**Comment by richardliaw** [comment]
(I'm rebranding this issue for Ray Tune + RaySGD placement group integration)

A current draft of this is here, though I think it is still not done: https://github.com/ray-project/ray/pull/11061/files

cc @oliverhu

**Comment by oliverhu** [comment]
👍  

Placement group is a very good to have for SGD (at least for tf) to avoid starvation and deadlock.

**Comment by oliverhu** [comment]
@richardliaw  @ericl  let's close this ticket ?
